### PR TITLE
foreach in all plays

### PR DESCRIPTION
**Pull request by hdemusg**
## Description
Replaced repetitive statements in plays with foreach loops

## Steps to test
Run simulator

Expected result: It should work

### CODE REVIEW

File: rj_gameplay/rj_gameplay/play/basic122.py [hunk 1]
```diff
@@ -81,6 +81,15 @@ def tick(
 
         # Get list of all skills with assigned roles from tactics
 
+        skills = []
+        skill_dict = {}
+
+        for s in role_requests.keys():
+            skills += s.tick(world_state, role_results[s])
+            skil_dict.update(role_results[s])
```
Comment (kfu02) [inline rj_gameplay/rj_gameplay/play/basic122.py]: fix this typo please

**Review by kfu02 — CHANGES_REQUESTED** [review]
doesn't build, please merge in the latest changes from ros2 that fix this

**Review by adithya-gv — CHANGES_REQUESTED** [review]
Code compiles and runs. Fix all of the GH Actions errors but otherwise, looks good to me.

**Review by anshulm07 — CHANGES_REQUESTED** [review]
Looks good to me, but GitHub Actions errors need to be fixed.

**Comment by kfu02** [comment]
superseded by https://github.com/RoboJackets/robocup-software/pull/1811

**Pull request closed without merging**